### ISSUE TITLE
[bug] Exception is assigned to variable

### ISSUE THREAD

**Issue by haering** [codingseb/ExpressionEvaluator#81]
Hi,

I found the bug, that if I assign the function result to a variable an exception thrown by that function will be assigned to that variable. 

In the following example I would expect the Evaluate function to throw the exception and the result variable does not exist, but in the current version the exception is thrown and additionally assigned to the variable.

```
[Test]
public void Evaluate_WithException_ThrowsExceptionAndDoesNotAssignIt()
{
    TestConnectExpressionEvaluator evaluator = new TestConnectExpressionEvaluator();
    
    evaluator.Variables.Add("dummy", new DummyForException());
    Assert.Throws<ExpressionEvaluatorSyntaxErrorException>(() => evaluator.Evaluate("result = dummy.bang()"));
    
    Assert.That(evaluator.Variables.ContainsKey("result"), Is.False);
}

public class DummyForException
{
  public void bang()
  {
      throw new Exception();
  }
}
```

**Comment by codingseb** [comment]
Hello @haering thanks for reporting this bug.
I think I know why I will correct it as soon as I can.

**Comment by codingseb** [comment]
I just published version 1.4.19.0 that correct this bug.
Have a nice day

**Comment by haering** [comment]
Thanks for your fast response and your work on this project.
